universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -135,44 +151,65 @@ public ContainerTag blockOrderPanel(ProgramDefinition program, long focusedBlock
                 Styles.W_1_5,
                 Styles.BORDER_R,
                 Styles.BORDER_GRAY_200);
+    addBlocksToContainer(
+        ret, program, program.getBlockDefinitions(Optional.empty()), focusedBlockId, 1);
+    ret.with(
+        submitButton("Add Block")
+            .withId("add-block-button")
+            .attr(Attr.FORM, CREATE_BLOCK_FORM_ID)
+            .withClasses(Styles.M_4));
+    return ret;
+  }
 
-    for (BlockDefinition block : program.blockDefinitions()) {
+  private void addBlocksToContainer(
+      ContainerTag container,
```

Comment:
Suggestion (if-minor): Would you make this function have no side effect, i.e. take in immutable parameters and return a new object? I believe this is more readable. Having a separate method for recursive calls that call this function in each level might make it easier.

Reply:
That looks doable since this method could create the container as a plain div and the consumers of this method can add whatever else they want to it after it returns.

Reply:
done